fastly-compute/geoip/src/index.js

```diff
@@ -0,0 +1,45 @@
+/// <reference types="@fastly/js-compute" />
+
+import { getGeolocationForIpAddress } from "fastly:geolocation"
+import { ConfigStore } from "fastly:config-store";
+
+async function app(event) {
+  try {
+   const configStore = new ConfigStore('strings')
+   const token = await configStore.get('token')
+
+    // If the value doesn't exist or is false, return Unauthorized
+   if (!token) {
+     let respBody = JSON.stringify({ Error: "Unauthorized" });
+     return new Response(respBody, {
+       status: 401,
+       headers: { 'Content-Type': 'application/json' }
+     });
+    }
+    let ip = new URL(event.request.url).searchParams.get('ip') || event.client.address
+    let geo = getGeolocationForIpAddress(ip);
+    let respBody = JSON.stringify({
+      geo: {
+        city: geo.city,
+        continent: geo.continent,
+        country_code: geo.country_code,
+        country_code3: geo.country_code3,
+        country_name: geo.country_name,
+        region: geo.region,
+      },
+});
+
+    return new Response( respBody, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  } catch (error) {
+    console.error(error);
+    return new Response("Internal Server Error", {
+      status: 500
+    });
+  }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏the entire function seems to be wrapped in the try/catch block.
Would Fastly respond with a 500 status if there were any errors without the try/catch? Do the function errors log to the admin console or something?

Basically, I'm asking whether we can simplify the outer behavior by letting their platform operate and return appropriate error response code is catastrophe occurs.

Reply:
This code format was retrieved from an example on [fastly's documentation,](https://developer.fastly.com/reference/api/dictionaries/dictionary/#:~:text=%7D%3B-,try%20%7B,%7D,-Show%20Less) and it seems should be reworked to better match what we are trying to produce.